File: HW1/HW1.java
import java.util.Arrays;
import java.util.Scanner;

public class HW1 {

public static String removeDigitsLower(String holder)
{
    int size = holder.length();
    String temp = "";

    for(int i = 0; i < size; i++)
    {
        if(!Character.isDigit(holder.charAt(i)))
        {
            temp = temp + holder.charAt(i);
        }
    }

    return temp;
}

public static Boolean palindrome(String holder) //assume removeDigitsLower is called on param
{
    int last = holder.length();
    for(int i = 0; i < holder.length(); i++)
    {
        if(holder.charAt(i) == holder.charAt(last-1-i))
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    return false;
}
public static void main(String[] args)
{
    Scanner holder = new Scanner(System.in);
    System.out.println("enter the string pls: ");
    String statement = holder.next();

    statement = removeDigitsLower(statement);
    Boolean pal = palindrome(statement);
    System.out.println("this statement is a palindrome: " + pal);

}
}

File: HW1/HW1.py
def remove_digits_lower(holder):
    temp = ""
    for char in holder:
        if not char.isdigit():
            temp += char
    return temp

def palindrome(holder): # assume remove_digits_lower is called on parameter
    last = len(holder)
    for i in range(len(holder)):
        if holder[i] == holder[last - 1 - i]:
            return True
        else:
            return False
    return False

def main():
    statement = input("Enter the string please: ")
    statement = remove_digits_lower(statement)
    pal = palindrome(statement)
    print("This statement is a palindrome:", pal)

if __name__ == "__main__":
    main()
